### PR TITLE
Update TensorRT-LLM

### PR DESCRIPTION
**Pull request by kaiyux**
* Model Support
  - Added DeepSeek V3/R1 support on PyTorch workflow.
* Features
  - LLM API
    - Added batched logits processor support.
    - Added EAGLE support.
  - Added FP8 quantization support for Qwen2-VL.
  - Added batched inference support for the LLM API MMLU example `examples/mmlu_llmapi.py`.
  - Added FP4 quantization-layernorm fusion plugin support. (Llama models only)
* API
  - Exposed `kc_cache_retention_config` from C++ `executor` API to the LLM API.

### CODE REVIEW

**Review by Shixiaowei02 — APPROVED** [review]

**Comment by JThh** [comment]
Hey @kaiyux, thanks for supporting DeepSeek V3.

May I know when the public release `v0.18.0` and its tensorrt-llm backend image (as in [here](https://catalog.ngc.nvidia.com/orgs/nvidia/containers/tritonserver/tags)) shall be available? 

Thanks.
